mixer/adapter/rbac/rbac.go

```diff
@@ -0,0 +1,176 @@
+	return
+}
+
+func (b *builder) SetAuthorizationTypes(types map[string]*authorization.Type) {}
+
+func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler, error) {
+	reg := store.NewRegistry(mixerconfig.StoreInventory()...)
+	s, err := reg.NewStore(b.adapterConfig.ConfigStoreUrl)
+	if err != nil {
+		env.Logger().Errorf("unable to connect to the configuration server: %v", err)
+		return nil, err
+	}
+	r := &configStore{}
+	err = startController(s, r, env)
+	if err != nil {
+		env.Logger().Errorf("unable to start controller: %v", err)
+		return nil, err
+	}
+
+	return &handler{rbac: r, env: env}, nil
+}
+
+// maxEvents is the likely maximum number of events
+// we can expect in a second. It is used to avoid slice reallocation.
+const maxEvents = 50
+
+var watchFlushDuration = time.Second
+
+// startController creates a controller from the given params.
+func startController(s store.Store, r *configStore, env adapter.Env) error {
+	data, watchChan, err := startWatch(s)
+	if err != nil {
+		env.Logger().Errorf("Error while starting watching CRDs: %v", err)
+		return err
+	}
+
+	c := &controller{
+		configState: data,
+		rbacStore:   r,
+	}
+
+	c.processRbacRoles(env)
+	// Start a goroutine to watch for changes on a channel and
+	// publishes a batch of changes via applyEvents.
+	env.ScheduleDaemon(func() {
+		// consume changes and apply them to data indefinitely
+		var timeChan <-chan time.Time
+		var timer *time.Timer
+		events := make([]*store.Event, 0, maxEvents)
+
+		for {
+			select {
+			case ev := <-watchChan:
+				if len(events) == 0 {
+					timer = time.NewTimer(watchFlushDuration)
+					timeChan = timer.C
+				}
+				events = append(events, &ev)
+			case <-timeChan:
+				timer.Stop()
+				timeChan = nil
+				env.Logger().Infof("Publishing %d events", len(events))
+				c.applyEvents(events, env)
+				events = events[:0]
+			}
+		}
+	})
+	return nil
+}
+
+// startWatch registers with store, initiates a watch, and returns the current config state.
+func startWatch(s store.Store) (map[store.Key]*store.Resource, <-chan store.Event, error) {
+	ctx := context.Background()
+
+	kindMap := make(map[string]proto.Message)
+	kindMap[serviceRoleKind] = &rbacproto.ServiceRole{}
+	kindMap[serviceRoleBindingKind] = &rbacproto.ServiceRoleBinding{}
+
+	if err := s.Init(ctx, kindMap); err != nil {
+		return nil, nil, err
+	}
+	// create channel before listing.
+	watchChan, err := s.Watch(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	return s.List(), watchChan, nil
+}
+
+////////////////// Request-time Methods //////////////////////////
+// authorization.Handler#HandleAuthorization
+func (h *handler) HandleAuthorization(ctx context.Context, inst *authorization.Instance) (adapter.CheckResult, error) {
+	s := status.OK
+	result, err := h.rbac.CheckPermission(inst, h.env)
+	if !result || err != nil {
+		s = status.WithPermissionDenied("RBAC: permission denied.")
+	}
+	return adapter.CheckResult{
+		Status: s,
```

Comment:
You should be setting the cacheability info in the result, otherwise the result will not be cached by the proxy and/or mixer.

Reply:
I added the cacheability info in the checkResult. Thanks for the instruction.